File: Leetcode/Day 4/Java/Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap <Integer, Integer> hmap = new HashMap<>(); 
        for (int i = 0; i < nums.length; i++){
            int k = target - nums[i];

            if (hmap.containsKey(k)){
                return new int[]{hmap.get(k), i};
            }
            else{
                hmap.put(nums[i], i);
            }
        }
        return new int[]{};
    }
}

File: Leetcode/Day 4/Python/Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        hashmap = {}
        res = []
        for i in range(len(nums)):
            value = target - nums[i]

            if value in hashmap:
                res += [nums.index(value), i]

            else:
                hashmap[nums[i]] = value

        return res
